modular_skyrat/modules/clock_cult/code/structures/traps/senders/delay.dm

```diff
@@ -0,0 +1,47 @@
+/obj/item/wallframe/clocktrap/delay
+	name = "clockwork timer"
+	desc = "A small, detached timer."
+	icon_state = "delayer"
+	result_path = /obj/structure/destructible/clockwork/trap/delay
+	clockwork_desc = "A device that can be attached to walls. When input is received, it will send an output signal a configurable (with multitool) time later."
+
+
+/obj/structure/destructible/clockwork/trap/delay
+	name = "clockwork timer"
+	desc = "A small timer attatched to the wall."
+	icon_state = "delayer"
+	component_datum = /datum/component/clockwork_trap/delay
+	unwrench_path = /obj/item/wallframe/clocktrap/delay
+	max_integrity = 15
+	clockwork_desc = "When input is received, it will send an output signal a configurable (with multitool) amount of time later."
+	/// How much time the signal is delayed
+	var/delay_time = 1 SECONDS
+
+
+/obj/structure/destructible/clockwork/trap/delay/multitool_act(mob/living/user, obj/item/tool)
+	delay_time = tgui_input_number(user, "Input delay time", "Clockwork Timer", 1 SECONDS, 120 SECONDS, 1 SECONDS)
```

Comment:
I'd probably just very quickly do a sanitization check on this, just to be safe.

Reply:
tgui_input_number prevents anything bad being sent